plasmapy/utils/data/downloader.py

```diff
@@ -0,0 +1,75 @@
+"""
+Contains functionality for downloading files from a URL. Intended for
+downloading files from the PlasmaPy data repository.
+
+"""
+
+import os
+import requests
+
+from urllib.parse import urljoin
+
+__all__ = ["get_file"]
+
+# Note: GitHub links have a problem where the Content-Encoding is set to
+# 'gzip' but the file is not actually compressed. This header is just ignored
+# by the get_file function.
+_BASE_URL = "https://raw.githubusercontent.com/PlasmaPy/PlasmaPy-data/main/"
+
+# TODO: use a config file variable to allow users to set a location for this folder?
+_DOWNLOADS_PATH = os.path.join(os.path.dirname(__file__), "downloads")
+
+
+def get_file(basename, base_url=_BASE_URL, directory=None):
+    r"""
+    Downloads a file from a URL (if the file does not already exist) and
+    returns the full local path to the file.
+
+    Parameters
+    ----------
+    basename : str
+        Name of the file to be downloaded (extension included).
+
+    base_url : str, optional
+        The base URL of the file to be downloaded. Defaults to the main
+        directory of `PlasmaPy's data repository`_.
+
+    directory : str, optional
+        The full path to the desired download location. Defaults to the
+        default PlasmaPy data download directory
+        plasmapy/utils/data/downloads/
```

Comment:
Not sure I understand the issue with root vs non-root, but this does sound nicer than keeping that data in the module, so I've made this change. 

Note that it isn't tested because the tests save data to a temporary directory...